common/amount.c

```diff
@@ -33,6 +33,9 @@ const char *fmt_amount_msat_btc(const tal_t *ctx,
 				const struct amount_msat *msat,
 				bool append_unit)
 {
+	if (msat->millisatoshis == 0)
+		return tal_fmt(ctx, append_unit ? "0btc" : "0");
```

Comment:
same as other comment, no need for `tal_fmt`

Reply:
But we promise to return a tal-allocated value on `ctx` here ?

Reply:
yeah, caller might tal_free it...